### ISSUE TITLE
How to report a command has completely been discarded, without any resulting event?

### ISSUE THREAD

**Issue by enikao** [LionWeb-io/specification#314]
Assume a client sends a `moveChild` command, and the repo acknowledges it. Thus, the client stores the command id to correlate it with the completion event.
No other commands are issued.
After some time, the repository concluded it cannot move the child, e.g. because it would create a containment loop.
How to report back to the client that no action was taken on that command? Do we need a "no-event event"?

**Comment by dslmeinte** [comment]
I don't think we need one, assuming that command IDs don't have any meaningful structure, such as being in sequence. An event needs to say which command it originated from (by ID), but not every command needs to result in an event. In CQRS, a command leads to 0..* events.

**Comment by enikao** [comment]
The reason for having an event is that the client can remove the command id from its list of outstanding commands. Otherwise that list might grow infinitely.

**Comment by dslmeinte** [comment]
OK, you've just answered the question with "yes" then 😁

**Comment by enikao** [comment]
**Decision: Introduce no-op event**

Rationale: Client needs to know it command has been taken into account, even if it had no effect.
